src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/JsonConverters/ParticleSystemJsonConverter.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace StateRecorder.JsonConverters
+{
+    public class ParticleSystemJsonConverter : Newtonsoft.Json.JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
```

Comment:
Most of those objects are serializable as they are.. despite a gajillion warnings about deprecation and other thread access control issues.  These custom converters exist though because.
1. I am only pulling the fields specific to that type you would see in the editor, not the entire nested behaviour state fields
2. Similar to #1 Some of Unity's types serialize with a bunch of extra fields you don't need to recreate / understand their state.  Vectors, Bounds, etc are examples where they have a lot of public fields and properties exposed, but you really only need a tiny subset of those to have the 'state'.  The other fields/properties are derivative values of the core state